pkg/controller/utils/certs.go

```diff
@@ -56,35 +56,37 @@ func GetSecret(ctx context.Context, client client.Client, name string, ns string
 
 // EnsureCertificateSecret ensures that the certificate in the
 // secret has the expected DNS names. If no secret is provided, a new
-// secret is created and returned. If the secret does have the
-// right DNS names then the secret is returned.
+// secret is created and returned alongside a true boolean. If the secret does have the
+// right DNS names then the secret is returned alongside a false boolean.
 // If the cert in the secret has invalid DNS names and the secret is operator
-// managed, then a new secret is created and returned. Otherwise,
+// managed, then a new secret is created and returned alongside a true boolean. Otherwise,
 // if the secret is user-supplied, an error is returned.
-func EnsureCertificateSecret(secretName string, secret *corev1.Secret, keyName string, certName string, certDuration time.Duration, svcDNSNames ...string) (*corev1.Secret, error) {
+func EnsureCertificateSecret(secretName string, secret *corev1.Secret, keyName string, certName string, certDuration time.Duration, svcDNSNames ...string) (*corev1.Secret, bool, error) {
```

Comment:
I like this change to return bool if the secret is created or not, but I think the comment could be improved.
I'd suggest something like
```
EnsureCertificateSecret ensures that the certificate in the
secret has the expected DNS names.
The first returned field (*corev1.Secret) is the validated or created Secret to use.
The second returned field (bool) is true if the Secret returned was generated and needs to be created, otherwise false is returned.
The third field (error) is nil if the Secret pass in is valid or was created successfully. If there was a problem creating the certificate or the Secret has invalid DNS names and the secret is not operator managed, an error is returned.
```
I think this is more readable.

Reply:
Function description updated

Reply:
Function description updated